tests/components/lutron_caseta/test_config_flow.py

```diff
@@ -158,3 +168,341 @@ async def test_duplicate_bridge_import(hass):
+        result["flow_id"],
+        {
+            CONF_HOST: "1.1.1.1",
+        },
+    )
+    await hass.async_block_till_done()
+    assert result2["type"] == "form"
+    assert result2["step_id"] == "link"
+
+    with patch.object(Smartbridge, "create_tls") as create_tls, patch(
+        "homeassistant.components.lutron_caseta.async_setup", return_value=True
+    ), patch(
+        "homeassistant.components.lutron_caseta.async_setup_entry",
+        return_value=True,
+    ):
+        create_tls.return_value = MockBridge(can_connect=True)
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {},
+        )
+        await hass.async_block_till_done()
+
+    assert result3["type"] == "create_entry"
+    assert result3["title"] == "1.1.1.1"
+    assert result3["data"] == {
+        CONF_HOST: "1.1.1.1",
+        CONF_KEYFILE: "lutron_caseta-1.1.1.1-key.pem",
+        CONF_CERTFILE: "lutron_caseta-1.1.1.1-cert.pem",
+        CONF_CA_CERTS: "lutron_caseta-1.1.1.1-ca.pem",
+    }
+
+
+async def test_zeroconf_host_already_configured(hass, tmpdir):
+    """Test starting a flow from discovery when the host is already configured."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    hass.config.config_dir = await hass.async_add_executor_job(
+        tmpdir.mkdir, "tls_assets"
+    )
+
+    config_entry = MockConfigEntry(domain=DOMAIN, data={CONF_HOST: "1.1.1.1"})
+
+    config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data={
+            CONF_HOST: "1.1.1.1",
+            ATTR_HOSTNAME: "lutron-abc.local.",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+
+async def test_zeroconf_lutron_id_already_configured(hass):
+    """Test starting a flow from discovery when lutron id already configured."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, data={CONF_HOST: "4.5.6.7"}, unique_id="abc"
+    )
+
+    config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data={
+            CONF_HOST: "1.1.1.1",
+            ATTR_HOSTNAME: "lutron-abc.local.",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+    assert config_entry.data[CONF_HOST] == "1.1.1.1"
+
+
+async def test_zeroconf_not_lutron_device(hass):
+    """Test starting a flow from discovery when it is not a lutron device."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    config_entry = MockConfigEntry(
```

Comment:
There is no real reason to create a config entry here since there is no interaction between the discovery and the existing config entry

Reply:
Good call, removed. 👍